cylc/flow/network/schema.py

```diff
@@ -276,6 +278,16 @@ class SortArgs(InputObjectType):
     return await resolvers.get_workflows(args)
 
 
+async def subscribe_workflows(root, info, **args):
+    while True:
+        args['workflows'] = [parse_workflow_id(w_id) for w_id in args['ids']]
+        args['exworkflows'] = [parse_workflow_id(w_id) for w_id in
+                               args['exids']]
+        resolvers = info.context.get('resolvers')
+        yield await resolvers.get_workflows(args)
```

Comment:
This is pretty much the same as `get_workflows` - used for a Query.

With a difference that we return the coroutine here.

Reply:
We could move the argument parsing to a common function called by both.

Reply:
Yes, all if it’s worth it... but if we move argument parsing to resolvers.py it would serve the same purpose... Then the schema would be pure schema

Reply:
>Then the schema would be pure schema

Ah, I think I'm starting to understand it. This would simplify the schema a bit more. Would it be alright to move this to a follow-up PR?

Reply:
Yes, it also might open other avenues, hiding the nuts and bolts 🔩 of the driving data... Like defining the schema in GraphQL pseudo code, like a document which is not implementation language specific  (which is a thing)